kafka_consumer/datadog_checks/kafka_consumer/new_kafka_consumer.py

```diff
@@ -82,12 +89,27 @@ def check(self):
                 self._context_limit,
             )
 
+        self._save_broker_timestamps()
+
         # Report the metrics
         self._report_highwater_offsets(self._context_limit)
         self._report_consumer_offsets_and_lag(self._context_limit - len(self._highwater_offsets))
 
         self._collect_broker_metadata()
 
+    def _load_broker_timestamps(self):
+        """Loads broker timestamps from persistent cache."""
+        self._broker_timestamps = defaultdict(dict)
+        try:
+            for topic_partition, content in json.loads(read_persistent_cache(self._broker_timestamp_cache_key)).items():
+                for offset, timestamp in content.items():
+                    self._broker_timestamps[topic_partition][int(offset)] = timestamp
+        except Exception:
+            self.log.warning('Could not read broker timestamps from cache')
```

Comment:
could you include the original exception in the message too so it gives some information about what the issue is?

Reply:
Makes sense. Done.